src/python/pants/util/collections_backport.py

```diff
@@ -0,0 +1,26 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from future.utils import PY2, PY3
+
+
+"""Custom backport of collections, due to limitations with future.moves.
```

Comment:
It would be good to put almost all of what you put in the PR description in this file, I think.